### DISCUSSION TITLE
Doesnt it make sense for `useIsFocused` to be false if the user is in the background?

### DISCUSSION THREAD

**Discussion by itsramiel** [react-navigation/react-navigation#12316] [Q&A] [answered]
Hi,

I would like to play some kind of animation when the user is on the screen, and useIsFocused hook comes to mind. The problem is that it doesnt change to false when the user goes to the background.

Wouldnt it make more sense if it returned false since the user is not focused anymore on the screen?

**Comment by satya164** [comment]
What do you mean by user is on the screen/goes to background?

**Reply by itsramiel** [reply]
So if the user has the app in the foregound, then puts the app in the background then the `useIsFocused` hook is still true. Would it make more sense to switch to false?

**Reply by satya164** [reply, marked as answer]
`useIsFocused` is for screen's focus status. if you want to also consider app's state you can make a hook that uses `useIsFocused` and `AppState` together.

**Reply by itsramiel** [reply]
As react navigation is more about screens that makes sense. Alright thank you

**Reply by itsramiel** [reply]
If anyone is interested in a hook which returns if the user is focused on the screen and app state is active:
```ts
import { AppState } from 'react-native';
import { useCallback, useEffect, useState } from 'react';
import { useNavigation } from '@react-navigation/native';

function getIsScreenActive(isScreenFocused: boolean) {
  return isScreenFocused && AppState.currentState === 'active';
}

export function useIsScreenActive() {
  const navigation = useNavigation();

  const [isScreenActive, setIsScreenActive] = useState(() =>
    getIsScreenActive(navigation.isFocused())
  );

  const updateIsScreenActive = useCallback(() => {
    setIsScreenActive(getIsScreenActive(navigation.isFocused()));
  }, [navigation, setIsScreenActive]);

  useEffect(() => {
    const unsubscribeFocusedListener = navigation.addListener(
      'focus',
      updateIsScreenActive
    );

    const unsubscribeBlurredListener = navigation.addListener(
      'blur',
      updateIsScreenActive
    );

    const appStateListener = AppState.addEventListener(
      'change',
      updateIsScreenActive
    );

    return () => {
      unsubscribeFocusedListener();
      unsubscribeBlurredListener();
      appStateListener.remove();
    };
  }, [navigation, setIsScreenActive]);

  return isScreenActive;
}
```
